src/service-worker/core.js

```diff
@@ -624,6 +624,10 @@ export function handleFetch(request, maybeClientId) {
       // If not, let's fetch and cache the request.
       return fetchJsFile(cache, versionedRequest, pathname, version);
     });
+  }).catch(err => {
+    // Throw error out of band.
+    Promise.reject(err);
```

Comment:
Doesn't this line just create a rejected promise? I thought you'd need to call the `throw` async to avoid having the promise swallow the error, e.g. with `setTimeout`.